src/core/InstrumentFunctions.cpp

```diff
@@ -408,7 +408,37 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 			continue;
 		}
 
-		const int dir = m_arpDirectionModel.value();
+		// Skip notes randomly
+		if( m_arpSkipModel.value() )
+		{
+
+			if( 101 * ( (float) rand() / (float) RAND_MAX ) < m_arpSkipModel.value() )
+			{
+				if( cur_arp_idx == 0 )
+				{
+					_n->setMasterNote();
+				}
+				// update counters
+				frames_processed += arp_frames;
+				cur_frame += arp_frames;
+				continue;
+			}
+		}
+
+		int dir = m_arpDirectionModel.value();
+
+		// Miss notes randomly. We intercept int dir and abuse it
+		// after need.  :)
+
+		int miss = m_arpMissModel.value();
+		if( miss )
+		{
+			if( 100 * ( (float) rand() / (float) RAND_MAX ) < ( miss * miss ) / 100 )
```

Comment:
It should be `RAND_MAX + 1` here to get 100% miss rate. Why `miss * miss`? This is not a percentage.